app/controllers/user_controllers.js

```diff
@@ -24,16 +90,27 @@ exports.buildUser = (req, res, next) => {
         email: req.body.email
       }
     : {};
-  User.create(newUser).then(
-    result => {
-      res.send(`Successfully logged in. Welcome, ${result.firstName} ${result.lastName}!`);
-    },
-    e => {
-      let errorMessage = '';
-      e.errors.forEach(error => {
-        logger.info(error.message);
+
+  let valMsg = validateUser(newUser);
+  if (valMsg.length > 0) {
+    valMsg = valMsg.slice(0, -1);
+    next(errors.defaultError(valMsg));
+  } else {
+    bcrypt
+      .hash(newUser.password, saltRounds)
+      .then(hash => {
+        newUser.password = hash;
+
+        User.createNewUser(newUser)
```

Comment:
Same here

Reply:
Where would the return statement go in here?

Reply:
Controller function (this is for our standard) should return a promise (if use one), if you add only a return in this line https://github.com/wolox-training/fj-express-js/blob/f4b4e69a57d3025272858ad60c627408acda7ebd/app/controllers/user_controllers.js#L36 and not here you are skipping this promise (ask me if you still don't understand)